Tests/ApolloTests/StarWars/API.swift

```diff
@@ -323,6 +384,15 @@ public final class HeroDetailsQuery: GraphQLQuery {
     "  }" +
     "}"
 
+  public static let selectionSet: [Selection] = [
+    Field("hero", arguments: ["episode": Variable("episode")], type: .object(Data.Hero.self), selectionSet: [
+      Field("__typename", type: .nonNull(.scalar(String.self))),
+      Field("name", type: .nonNull(.scalar(String.self))),
+      FragmentSpread(Data.Hero.AsHuman.self),
```

Comment:
It looks like you’re generating a fragment here for inline fragments. What happens when you have an inline fragment without a type condition?

```graphql
{
  a
  b
  ... {
    c
    d
  }
}
```

I don’t know anyone who uses inline fragments without a type condition, but it is technically possible. Theoretically I can see them being useful with directives, and in the future we may use them in Apollo Client JS to allow users to specify fields to partially read from the store. Say the user _always_ wanted fields `a` and `b`, but only wanted fields `c` and `d` if they were available in the store and if they are not available then the user just wants `a` and `b` without an error being thrown.